otx/core/config/manager.py

```diff
@@ -0,0 +1,137 @@
+    * Write the data to the workspace
+    * Write the data configuration to the workspace
+
+    However, it will supports lots of things in the near future.
+    * Automatic train type detection (Supervised, Self, Semi)
+    * Automatic resource allocation (num_workers, HPO)
+
+    """
+
+    def __init__(self):
+        # Currently, Datumaro.auto_split() can support below 3 tasks
+        # Classification, Detection, Segmentation
+        self.task_data_dict = {
+            "CLASSIFICATION": ["imagenet"],
+            "DETECTION": ["coco", "voc", "yolo"],
+            "SEGMENTATION": ["common_semantic_segmentation", "voc", "cityscapes", "ade20k2017", "ade20k2020"],
+            # "ACTION_CLASSIFICATION": ["multi-cvat"],
+            # "ACTION_DETECTION": ["multi-cvat"],
+            # "ANOMALY_CLASSIFICATION": ["mvtec"],
+            # "ANOMALY_DETECTION": ["mvtec"],
+            # "ANOMALY_SEGMENTATION": ["mvtec"],
+            # "INSTANCE_SEGMENTATION": ["coco", "voc"],
+        }
+        self.data_format: str = ""
+        self.splitted_dataset: Dict[str, IDataset]
+
+    def auto_task_detection(self, train_data_roots: str) -> str:
+        """Detect task type automatically."""
+        data_format = self._get_data_format(train_data_roots)
+        return self._get_task_type(data_format)
+
+    def _get_data_format(self, data_root: str) -> str:
+        """Get dataset format."""
+        self.data_format = DatasetManager.get_data_format(data_root)
+        return self.data_format
+
+    def _get_task_type(self, data_format: str) -> str:
+        """Detect task type.
+
+        For some datasets (i.e. COCO, VOC, MVTec), can't be fully automated.
+        Because those datasets have several format at the same time.
+        (i.e. for the COCO case, object detection and instance segmentation annotations coexist)
+        In this case, the task_type will be selected to default value.
+
+        For action tasks, currently action_classification is default.
+
+        If Datumaro supports the Kinetics, AVA datasets, MVTec, _is_cvat_format(), _is_mvtec_format()
+        functions will be deleted.
+        """
+
+        task = ""
+        # pick task type
+        for task_key, data_value in self.task_data_dict.items():
+            if data_format in data_value:
+                task = task_key
+        return task
+
+    def auto_split_data(self, train_data_root: str, train_task: str):
+        """Automatically Split train data --> train/val dataset."""
+        train_data_format = DatasetManager.get_data_format(train_data_root)
+        train_dataset = DatasetManager.import_dataset(data_root=train_data_root, data_format=train_data_format)
+        self.splitted_dataset = DatasetManager.auto_split(
+            task=train_task,
+            dataset=DatasetManager.get_train_dataset(train_dataset),
+            split_ratio=[("train", 0.8), ("val", 0.2)],
+        )
+        print("[*] Auto-split enabled.")
+
+    def write_data_with_cfg(
```

Comment:
How about not generating `data.yaml` in `builder.py` and always managing it here?

Here, it seems good to create a `data.yaml` containing `train_data_roots` or `val_data_roots` if they are updated, and create an empty `data.yaml` if they don't exist.